src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,376 @@
+  const adjustedHour = hour % 12;
+  return period === 'PM' ? adjustedHour + 12 : adjustedHour;
+};
+
+const formatHourTo12 = (hour: number) => {
+  const adjustedHour = hour % 12;
+  return adjustedHour === 0 ? 12 : adjustedHour;
+};
+
+export type TimePickerProps = {
+  /**
+   * Selected date.
+   */
+  date?: Date;
+  /**
+   * Callback when date is changed.
+   */
+  onChange?: (date: Date) => void;
+  /**
+   * Format of the time. 12h or 24h.
+   * @default false
+   */
+  use12Hours?: boolean;
+  /**
+   * Precision of the time.
+   * @default 'minutes'
+   */
+  precision?: 'hours' | 'minutes' | 'seconds';
+  /**
+   * Change step of the hours displayed.
+   * @default 1
+   */
+  hourStep?: number;
+  /**
+   * Change step of the minutes displayed.
+   * @default 1
+   */
+  minuteStep?: number;
+  /**
+   * Change step of the seconds displayed.
+   * @default 1
+   */
+  secondStep?: number;
+};
+
+/**
+ * Time picker component
+ * @example
+ * <TimePicker date={new Date()} onChange={(e) => console.log('New time value: ' + e)} />
+ */
+export const TimePicker = (props: TimePickerProps): JSX.Element => {
+  const {
+    date,
+    onChange,
+    use12Hours = false,
+    precision = 'minutes',
+    hourStep = 1,
+    minuteStep = 1,
+    secondStep = 1,
+  } = props;
+
+  useTheme();
+
+  const [selectedTime, setSelectedTime] = React.useState(date);
+  const [focusedTime, setFocusedTime] = React.useState(
+    selectedTime ?? new Date(),
+  );
+  const [period, setPeriod] = React.useState<'PM' | 'AM' | undefined>(
+    use12Hours ? (focusedTime?.getHours() > 11 ? 'PM' : 'AM') : undefined,
+  );
+
+  React.useEffect(() => {
+    setFocusedTime(date ?? new Date());
+    setSelectedTime(date);
+  }, [date]);
+
+  const onHourClick = (hour: number) => {
+    const adjustedHour = use12Hours ? formatHourFrom12(hour, period) : hour;
+    const adjustedSelectedTime = selectedTime ?? new Date();
+    adjustedSelectedTime.setHours(adjustedHour);
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const onMinutesClick = (minutes: number) => {
+    const adjustedSelectedTime = selectedTime ?? new Date();
+    adjustedSelectedTime.setMinutes(minutes);
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const onSecondsClick = (seconds: number) => {
+    const adjustedSelectedTime = selectedTime ?? new Date();
+    adjustedSelectedTime.setSeconds(seconds);
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const onPeriodClick = (value: string) => {
+    const adjustedSelectedTime = selectedTime ?? new Date();
+    const currentHours = adjustedSelectedTime.getHours();
+    if (value === 'AM' && currentHours > 11) {
+      setPeriod(value);
+      adjustedSelectedTime.setHours(currentHours - 12);
+    }
+    if (value === 'PM' && currentHours <= 12) {
+      setPeriod(value);
+      adjustedSelectedTime.setHours(currentHours + 12);
+    }
+    updateCurrentTime(adjustedSelectedTime);
+  };
+
+  const updateCurrentTime = (time: Date) => {
```

Comment:
We might need to look at the precision because currently if precision is minutes and I select 10:40, it can set like 10:40:34. If I am going to use that in filter in from field I would expect that 10:40:05 would be shown for me but with current implementation it would be not.

Reply:
Adjusting date now according to the precision on select.